src/flint/types/fmpz_mpoly.pyx

```diff
@@ -994,5 +1049,140 @@ cdef class fmpz_mpoly_vec:
+            return <bint>fmpz_mpoly_vec_is_groebner(self.val, NULL, self.ctx.val)
+        elif typecheck(other, fmpz_mpoly_vec):
+            self.ctx.compatible_context_check((<fmpz_mpoly_vec>other).ctx)
+            return <bint>fmpz_mpoly_vec_is_groebner(self.val, (<fmpz_mpoly_vec>other).val, self.ctx.val)
+        else:
+            raise TypeError(f"expected either None or a fmpz_mpoly_vec, got {type(other)}")
+
+    def is_autoreduced(self) -> bool:
+        """
+        Check if self is auto-reduced (or inter-reduced).
+
+            >>> from flint import Ordering
+            >>> ctx = fmpz_mpoly_ctx.get_context(2, Ordering.lex, nametup=('x', 'y'))
+            >>> x, y = ctx.gens()
+            >>> f2 = 3*x**2 - y
+            >>> k = x*y - x
+            >>> k2 = 3*x*y - 3*x
+            >>> h = y**2 - y
+            >>> vec = fmpz_mpoly_vec([f2, k, h], ctx)
+            >>> vec.is_autoreduced()
+            True
+            >>> vec = fmpz_mpoly_vec([f2, k2, h], ctx)
+            >>> vec.is_autoreduced()
+            False
+
+        """
+        return <bint>fmpz_mpoly_vec_is_autoreduced(self.val, self.ctx.val)
+
+    def autoreduction(self, groebner=False) -> fmpz_mpoly_vec:
+        """
+        Compute the autoreduction of `self`. If `groebner` is True and `self` is a Gröbner basis, compute the reduced
+        reduced Gröbner basis of `self`, throws an `RuntimeError` otherwise.
+
+            >>> from flint import Ordering
+            >>> ctx = fmpz_mpoly_ctx.get_context(2, Ordering.lex, nametup=('x', 'y'))
+            >>> x, y = ctx.gens()
+            >>> f2 = 3*x**2 - y
+            >>> k2 = 3*x*y - 3*x
+            >>> h = y**2 - y
+            >>> vec = fmpz_mpoly_vec([f2, k2, h], ctx)
+            >>> vec.is_autoreduced()
+            False
+            >>> vec2 = vec.autoreduction()
+            >>> vec2.is_autoreduced()
+            True
+            >>> vec2
+            fmpz_mpoly_vec([3*x^2 - y, x*y - x, y^2 - y], ctx=fmpz_mpoly_ctx(2, '<Ordering.lex: 0>', ('x', 'y')))
+
+        """
+
+        cdef fmpz_mpoly_vec h = fmpz_mpoly_vec(0, self.ctx)
+
+        if groebner:
+            if not self.is_groebner():
+                raise RuntimeError("reduced Gröbner basis construction requires that `self` is a Gröbner basis.")
+            fmpz_mpoly_vec_autoreduction_groebner(h.val, self.val, self.ctx.val)
+        else:
+            fmpz_mpoly_vec_autoreduction(h.val, self.val, self.ctx.val)
+
+        return h
+
+    def buchberger_naive(self, limits=None):
+        """
+        Compute the Gröbner basis of `self` using a naive implementation of Buchberger’s algorithm.
+
+        Provide `limits` in the form of a tuple of `(ideal_len_limit, poly_len_limit, poly_bits_limit)` to halt
+        execution if the length of the ideal basis set exceeds `ideal_len_limit`, the length of any polynomial exceeds
+        `poly_len_limit`, or the size of the coefficients of any polynomial exceeds `poly_bits_limit`.
```

Comment:
As a general point I find these 120 length lines hard to read. They wrap over in most views like here in the browser or my editor etc.

Even if I fullscreen so that it does not wrap I find it hard to read such long lines of text.

Reply:
Certainly, will make those changes. Just a consequence of me using a big monitor

Reply:
I've reflowed `fmpz_mpoly_vec` to 88 chars as an example